cmp/project.py

```diff
@@ -1014,197 +1037,192 @@ def run_individual(
+            exit_code = 0
+        else:  # pragma: no cover
+            print("ERROR : Invalid inputs for anatomical pipeline")
+            exit_code = 1
+            return exit_code
+
+    anat_valid_outputs, msg = anat_pipeline.check_output()
+    anat_pipeline.fill_stages_outputs()
+
+    return anat_pipeline, exit_code, anat_valid_outputs, msg
+
+
+def run_dmri_pipeline(project, bids_layout, anat_pipeline):
+    """Initialize and run diffusion pipeline based on `project.dmri_config_file` pipeline configuration, that has to be set a-priori."""
+    dmri_valid_inputs, dmri_pipeline = init_dmri_project(
+        project, bids_layout, False
+    )
+    if dmri_pipeline is not None:
+        dmri_pipeline.parcellation_scheme = anat_pipeline.parcellation_scheme
+        dmri_pipeline.atlas_info = anat_pipeline.atlas_info
+        if anat_pipeline.parcellation_scheme == "Custom":
+            dmri_pipeline.custom_atlas_name = anat_pipeline.stages["Parcellation"].config.custom_parcellation.atlas
+            dmri_pipeline.custom_atlas_res = anat_pipeline.stages["Parcellation"].config.custom_parcellation.res
+        if dmri_valid_inputs:
+            dmri_pipeline.process()
+            exit_code = 0
+        else:  # pragma: no cover
+            print("   ... ERROR : Invalid inputs for diffusion pipeline")
+            exit_code = 1
+            return exit_code
+
+        dmri_pipeline.fill_stages_outputs()
+    return exit_code
+
+
+def run_fmri_pipeline(project, bids_layout, anat_pipeline):
+    """Initialize and run fMRI pipeline based on `project.fmri_config_file` pipeline configuration, that has to be set a-priori."""
+    fmri_valid_inputs, fmri_pipeline = init_fmri_project(
+        project, bids_layout, False
+    )
+    if fmri_pipeline is not None:
+        fmri_pipeline.parcellation_scheme = anat_pipeline.parcellation_scheme
+        fmri_pipeline.atlas_info = anat_pipeline.atlas_info
+        fmri_pipeline.subjects_dir = anat_pipeline.stages["Segmentation"].config.freesurfer_subjects_dir
+        fmri_pipeline.subject_id = anat_pipeline.stages["Segmentation"].config.freesurfer_subject_id
+        if anat_pipeline.parcellation_scheme == "Custom":
+            fmri_pipeline.custom_atlas_name = anat_pipeline.stages["Parcellation"].config.custom_parcellation.atlas
+            fmri_pipeline.custom_atlas_res = anat_pipeline.stages["Parcellation"].config.custom_parcellation.res
+        print("Freesurfer subjects dir: {}".format(fmri_pipeline.subjects_dir))
+        print("Freesurfer subject id: {}".format(fmri_pipeline.subject_id))
+
+        if fmri_valid_inputs:
+            print(">> Process fMRI pipeline")
+            fmri_pipeline.process()
+            exit_code = 0
+        else:  # pragma: no cover
+            print("   ... ERROR : Invalid inputs for the fMRI pipeline")
+            exit_code = 1
+            return exit_code
+
+        fmri_pipeline.fill_stages_outputs()
+
+    return exit_code
```

Comment:
Here also, exit_code will be unassigned if fmri_pipeline is None

Reply:
Addressed by commit https://github.com/connectomicslab/connectomemapper3/pull/149/commits/832eb5bbb29c8b966b8beaf2c2709b71cdc79163